smithy-aws-protocol-tests/model/restJson1/documents.smithy

```diff
@@ -324,3 +324,73 @@ apply DocumentTypeAsPayload @httpResponseTests([
         }
     }
 ])
+
+/// This example serializes documents as the value of maps.
+@idempotent
+@http(uri: "/DocumentTypeAsMapKey", method: "PUT")
+operation DocumentTypeAsMapKey {
```

Comment:
Document is the map _value_ in these tests, all the names should be updated accordingly.